addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -688,6 +688,69 @@ publicWidget.registry.CarouselBootstrapUpgradeFix = publicWidget.Widget.extend({
     },
 });
 
+publicWidget.registry.CarouselSingleScrollMode = publicWidget.Widget.extend({
+    selector: ".carousel.o_slide_single",
+    disabledInEditableMode: false,
+    events: {
+        'slide.bs.carousel': '_onSlide',
+        'slid.bs.carousel': '_onSlideEnd',
+    },
+
+    /**
+     * Allow the carousel to cycle otherwise it would reach an end
+     * This method is called after when the carousel is sliding
+     *
+     * @private
+     */
+    _onSlide(e) {
+        const carousel = e.target;
+        const {carouselItems, totalItems, itemsPerSlide} = this._getCarouselData(carousel);
+
+        const relatedTarget = e.relatedTarget;
+        const idx =  Array.from(relatedTarget.parentNode.children).indexOf(relatedTarget);
+
+        // Check if we're near the end of the carousel and need to append items
+        if (idx >= totalItems - itemsPerSlide + (idx === 1)) {
+            const it = itemsPerSlide - (totalItems - idx) + 1;
+            for (let i = 0; i < it; i++) {
+                carouselItems[i].parentElement.appendChild(carouselItems[i]);
+            }
+        }
+    },
+
+    /**
+     * Prepare the carousel for the next sliding animation if needed
+     * This method is called after a sliding animation has finished
+     *
+     * @private
+     */
+    _onSlideEnd(e) {
+        const carousel = e.target;
+        const {carouselItems, totalItems, itemsPerSlide} = this._getCarouselData(carousel);
+
+        const relatedTarget = e.relatedTarget;
+        const idx = Array.from(relatedTarget.parentNode.children).indexOf(relatedTarget);
+
+        // Check if there is enough room for next sliding after animation
+        if (idx + itemsPerSlide >= totalItems) {
+            carouselItems[0].parentElement.appendChild(carouselItems[0]);
+        }
+    },
+
+    /**
+     * Helper method to get data about the current carousel
+     *
+     * @private
+     */
+    _getCarouselData(carousel) {
```

Comment:
For most of the variables I agree and I will change that. Unfortunately for the `carouselItemsEls` it has to be recomputed at every slide because the elements order change after every animation, If we put it in the start function the order is `this.carouselItemsEls` is not updated correctly